### ISSUE TITLE
Excluding any `filetype` or `buftype` doesn't work

### ISSUE THREAD

**Issue by turtureanu** [beauwilliams/focus.nvim#53]
### Description

I am trying to exclude any terminal windows from auto-resizing, the problem is that configuring `exclude_filetype` and/or `exclude_buftype` with the value of `"toggleterm"` doesn't work.

The `filetype` _and_ `buftype` of the terminal window are equal to null (`filetype=`, `buftype=`).

After manually changing the `filetype` and `buftype` using a [`TermOpen`](https://neovim.io/doc/user/autocmd.html#TermOpen) `autocmd` to `term`;
then using those to configure `focus.nvim` like so:

```lua
require("focus").setup({excluded_filetypes = {"term"}, excluded_buftype = "term"})
```

The `exclude_*` options still had no effect.

I also want to set an initial size for my terminal windows, normally I would do that using the following commands:

```vimscript
:botright:10new
:terminal
```

> Where `10` is the actual size.

But this also doesn't work as `focus.nvim` instantly auto-resizes the newly created window.

#### Expected behavior

Statically sized terminal window.

#### Actual behavior

`focus.nvim` default behavior

#### My current `focus.nvim` configuration

`init.lua`:

```lua
-- focus.nvim
require("focus").setup({compatible_filetrees = {"nvimtree"}, excluded_filetypes = {"toggleterm"}, excluded_buftypes = {"toggleterm"}})

-- Packer
return require("packer").startup(function()

  -- Packer can manage itself
  use "wbthomason/packer.nvim"
  
  -- focus.nvim
  use "beauwilliams/focus.nvim"
end)
```

**Comment by beauwilliams** [comment]
That is odd. I noticed that for toggleterm you must set the filetype exclusion to 'term'

You can test by opening ToggleTerm, hitting esc and running `:echo &ft`

I use toggleterm myself and have my plugin configured like so.

```lua
		use({
			'beauwilliams/focus.nvim',
	                cmd = { "FocusSplitNicely", "FocusSplitCycle" },
			module = "focus",
			config = function()
				require('focus').setup({
					tmux = false,
					hybridnumber = true,
					excluded_filetypes = { 'fterm', 'term' },
					signcolumn = 'number',
					bufnew = true,
				})
			end,
		})

```

There is a bug with packer I am seeing. I found that I need to save my *plugins*.lua file and run `PackerCompile` twice for changes to `config=function` stuff to have effect.

So I write the file, run packercompile, reopen vim, write file again, recompile packer, then reopen.. Seems to be only a packer thing however..

**Comment by turtureanu** [comment]
If I execute the following while the `filetype` is excluded, the initial size will be `10`:
```vimscript
:bo:10new|:terminal
```

Alas,  after switching to another window and back, `focus.nvim` will still resize the terminal window.

I did not use [toggleterm.nvim](https://github.com/akinsho/toggleterm.nvim).

After installing it, `excluded_filetypes = {"toggleterm"}` worked and I got the expected behavior.

> That is odd. I noticed that for toggleterm you must set the filetype exclusion to 'term'
>
> You can test by opening ToggleTerm, hitting esc and running :echo &ft

`:echo &ft` returns `toggleterm`, and changing `excluded_filetypes = {"toggleterm"}` to `excluded_filetypes = {"term"}` gets me the undesired behavior (the terminal window acts like any other window).
